Fixes silo logging + add log file based logging to silo

Gamer025 opened the pull request:
## About The Pull Request
Fixes deposit logs not working for the silo because they passed in the mats argument in the wrong format.
Also adds log file based logging to the silo with a new log file: silo.log.
This log should contain every deposit and withdraw action performed on the silo and from which machine and location inluding material amounts.
## Why It's Good For The Game
Fixes broken ingame logs + adds better external logs for potential stat tracking.
## Changelog
:cl: Gamer025
fix: Ingame ore silo logs should now log deposit actions correctly
config: New config for silo logs
/:cl:

Gamer025 committed:
Fix material deposit logging and break everything else

Gamer025 committed:
Actually not broken, improve logging text

Mothblocks commented on code/__DEFINES/logging.dm:
```diff
@@ -40,6 +40,7 @@
 #define LOG_VICTIM (1 << 19)
 #define LOG_RADIO_EMOTE (1 << 20)
 #define LOG_SPEECH_INDICATORS (1 << 21)
+#define LOG_SILO (1 << 22)
```
I don't think we need more of these. We're running out of room

Mothblocks commented on code/__HELPERS/logging/silo.dm:
```diff
@@ -0,0 +1,3 @@
+/proc/log_silo(text)
+	if (CONFIG_GET(flag/log_silo))
```
We don't need a config for every single logging entry

Gamer025 replied:
Made it so that silo logs are always created

Mothblocks reviewed (commented):
@ZephyrTFA Please advise because this feels like the type of thing that should be JSON logged if we got that merged

Gamer025 committed:
Removes config (always log) +
fix material amount wrongly logged in some cases +
document what kind of args the silo_log method function wants

ZephyrTFA commented:
Yeah, please convert this into using the json logging system.
See:
`code\modules\logging\`

You need to add a category, and thats about it really.

Gamer025 committed:
Rework logging to use new JSON based system

Gamer025 commented on code/__HELPERS/logging/silo.dm:
```diff
@@ -0,0 +1,2 @@
+/proc/log_silo(text)
```
Proc no longer exists

Gamer025 commented:
Reworked it to use the new JSON based system, PR should be good now

Mothblocks commented:
@ZephyrTFA Doesn't look like there's any version field for the ore silo? All new logs should have a version number set, right? Otherwise why do we have it?

ZephyrTFA commented:
> @ZephyrTFA Doesn't look like there's any version field for the ore silo? All new logs should have a version number set, right? Otherwise why do we have it? The default is 0.0.0 if I remember right, which implies unstable.

Semver isn't a thing yet. You're thinking of the general logging rewrite still waiting on json encode to be fixed

Mothblocks commented:
:( okay

Mothblocks commented:
(also json_encode has been fixed for a long time)
